Source: dleverton
Upstream: submitted by email
Reason: fix bash-4 -c 'printf -v foo %s bar; echo ${foo-unset}' and
        variations, which were breaking on amd64 (leaving foo set to
	the empty string)

--- bash-4.0-rc1/builtins/printf.def~	2009-01-04 19:32:23.000000000 +0000
+++ bash-4.0-rc1/builtins/printf.def	2009-01-18 22:50:45.000000000 +0000
@@ -879,16 +879,18 @@
 
   SH_VA_START (args, format);
   blen = vsnprintf (vbuf + vblen, vbsize - vblen, format, args);
+  va_end (args);
 
   nlen = vblen + blen + 1;
   if (nlen >= vbsize)
     {
       vbsize = ((nlen + 63) >> 6) << 6;
       vbuf = (char *)xrealloc (vbuf, vbsize);
+      SH_VA_START (args, format);
       blen = vsnprintf (vbuf + vblen, vbsize - vblen, format, args);
+      va_end (args);
     }
 
-  va_end (args);
   vblen += blen;
   vbuf[vblen] = '\0';
 
